lib/wallet/account/hot-wallet-private-key.ts

```diff
@@ -116,29 +120,50 @@ export class HotWalletPrivateKey implements IWallet {
       throw new Error("No transaction to sign");
     }
 
-    // Replace the input script with the provided script
-    if (options?.scriptHex) {
-      const replaceIndex = pending.transaction.inputs.findIndex(
-        (input) => input.signatureScript === "",
-      );
-      if (replaceIndex === -1) {
-        throw new Error("No input to replace");
+    const signed = await this.signTx(pending.transaction, options?.scripts);
+    return (await this.rpcClient.submitTransaction({ transaction: signed }))
+      .transactionId;
+  }
+
+  // NOTE: This method does not support signing with multiple keys
+  async signTx(tx: Transaction, scripts?: ScriptOption[]) {
+    if (scripts) {
+      for (const script of scripts) {
```

Comment:
Can be ran concurrently?

```typescript
      await Promise.all(
        scripts.map((script) => this.signTxInputWithScript(tx, script)),
      );

```

Reply:
Yes, it can.

Reply:
https://github.com/forbole/kastle/pull/22/commits/78f9525dcebec6b57c3361ac3cdb201e81bba0fc